docs/adrs/waitlists_relationships.md

```diff
@@ -0,0 +1,344 @@
+# Waitlists as Relationships
+
+* Status: approved
+* Deciders: <CTMS stakeholders> + <Basket stakeholders>
+* Date: November 25, 2022
+
+## Context and Problem Statement
+
+Onboarding new waiting lists requires changing, releasing, and redeploying both CTMS and Basket.
+
+Each waitlist may have its own schema with specific collected fields (eg. `country`, `platform`, ...).
+
+Basket uses [some specific code](https://github.com/mozmeao/basket/blob/341facbb2b199bfe2f26488942d0fa251010c1c8/basket/news/backends/ctms.py#L84-L87) to transform flat form data into CTMS nested objects, that has to be modified when a new waitlist has to be sent to CTMS.
+
+CTMS [stores the different waitlists as individual tables](https://github.com/mozilla-it/ctms-api/blob/3d0c0071c8f2565f1a24f039adc4f5526d58e0d2/ctms/models.py#L63-L66), with a relationship in the Email contact model. And in order to validate the submitted fields, each waitlist [has its own schema](https://github.com/mozilla-it/ctms-api/blob/3d0c0071c8f2565f1a24f039adc4f5526d58e0d2/ctms/schemas/vpn.py#L10-L32). If tomorrow we have to onboard 50 new waitlists, we would have to create 50 new tables in the database.
+
+Our current approach to onboard a new waiting lists obliges us:
+
+1. Open a pull-request to add the tables and schemas in CTMS
+1. Request and wait for review
+1. Request and wait for release
+1. Request to run database migrations
+1. Request and wait for deployment
+1. Open a pull-request to modify the CTMS mapping code in Basket
+1. Request and wait for review
+1. Request and wait for release
+1. Request and wait for deployment
+
+The amount of code to be modified in each step is reasonably low, but the amount of coordination and efforts to enable a new waiting list is too high.
+
+We are looking for a solution to streamline waitlists onboarding.
+
+## Decision Drivers
+
+In order to choose our solution we considered the following criteria:
+
+- **Complexity**: Low → High: how complex is the solution
+- **Cost**: Low → High: how much engineering efforts is required
+- **Onboarding Efforts**: Low → High: how much efforts are necessary
+
+## Considered Options
+
+1. [Option 0 - Do nothing](#option-0---do-nothing)
+1. [Option 1 - Store waitlists as relationships](#option-1---store-waitlists-as-relationships)
+1. [Option 2 - Option 1 + Generic waitlists forms in Basket](#option-2---option-1--generic-waitlists-forms-in-basket)
+1. [Option 3 - Option 2 + Waitlist schemas in database](#option-3---option-2--waitlist-schemas-in-database)
```

Comment:
very much a nitpick, but would it be possible to start the numbering at `0`? It's a little jarring to see that `2.` is option `1`.